py-polars/polars/io/excel/functions.py

```diff
@@ -249,70 +267,196 @@ def read_excel(
             sheet_id=sheet_id,
             sheet_name=sheet_name,
             read_csv_options=read_csv_options,
+            schema_overrides=schema_overrides,
             raise_if_empty=raise_if_empty,
         )
 
 
 def _initialise_excel_parser(
-    engine: str | None,
+    engine: Literal["xlsx2csv", "openpyxl", "odf"] | None,
     source: str | BytesIO | Path | BinaryIO | bytes,
     xlsx2csv_options: dict[str, Any],
 ) -> tuple[Any, Any, list[dict[str, Any]]]:
     """Instantiate the indicated Excel parser and establish related properties."""
     if isinstance(source, (str, Path)):
         source = normalize_filepath(source)
+    if engine is None and str(source).lower().endswith(".ods"):
+        engine = "odf"
 
-    if engine == "openpyxl":
+    if engine == "xlsx2csv" or engine is None:  # default
+        try:
+            import xlsx2csv
+        except ImportError:
+            raise ModuleNotFoundError(
+                "Required package not installed\n\nPlease run: `pip install xlsx2csv`"
+            ) from None
+        parser = xlsx2csv.Xlsx2csv(source, **xlsx2csv_options)
+        sheets = parser.workbook.sheets
+        return _read_excel_sheet_xlsx2csv, parser, sheets
+
+    elif engine == "openpyxl":
         try:
             import openpyxl
         except ImportError:
             raise ImportError(
-                "openpyxl is not installed\n\nPlease run `pip install openpyxl`"
+                "Required package not installed\n\nPlease run `pip install openpyxl`"
             ) from None
-        parser: openpyxl.Workbook = openpyxl.load_workbook(source, data_only=True)
-        sheets = [
-            {"index": i + 1, "name": sheet.title} for i, sheet in enumerate(parser)
-        ]
+        parser = openpyxl.load_workbook(source, data_only=True)
+        sheets = [{"index": i + 1, "name": ws.title} for i, ws in enumerate(parser)]
         return _read_excel_sheet_openpyxl, parser, sheets
 
-    elif engine == "xlsx2csv" or engine is None:  # default
+    elif engine == "odf":
         try:
-            import xlsx2csv
+            import ezodf
         except ImportError:
-            raise ModuleNotFoundError(
-                "xlsx2csv is not installed\n\nPlease run: `pip install xlsx2csv`"
+            raise ImportError(
+                "Required package not installed\n\nPlease run `pip install ezodf lxml`"
             ) from None
```

Comment:
We have used `dependencies` for core functionality that we think is likely to be loaded, but we want to defer the first load in case it isn't. Less usual/edge-case functionality that we think is unlikely to be used has tended to be import-guarded instead.

Reply:
I'll make an issue for determining what to do with optional dependencies. Doesn't have to block this PR.